pkg/operator/observe_config.go

```diff
@@ -90,69 +89,111 @@ func (c ConfigObserver) sync() error {
 	// don't worry about errors
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+
+	var errs []error
+	var observedConfigs []map[string]interface{}
+	for _, i := range rand.Perm(len(c.observers)) {
+		var currErrs []error
+		observedConfig, currErrs := c.observers[i](c.listers, currentConfig)
+		observedConfigs = append(observedConfigs, observedConfig)
+		errs = append(errs, currErrs...)
 	}
 
-	glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
-	operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
-	if _, err := c.operatorConfigClient.KubeControllerManagerOperatorConfigs().Update(operatorConfig); err != nil {
-		return err
+	mergedObservedConfig := map[string]interface{}{}
+	for _, observedConfig := range observedConfigs {
+		mergo.Merge(&mergedObservedConfig, observedConfig)
+	}
+
+	if !equality.Semantic.DeepEqual(currentConfig, mergedObservedConfig) {
+		glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, mergedObservedConfig))
+		operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: mergedObservedConfig}}
+		operatorConfig, err = c.operatorConfigClient.KubeControllerManagerOperatorConfigs().Update(operatorConfig)
```

Comment:
```suggestion
		updatedOperatorConfig, err := c.operatorConfigClient.KubeControllerManagerOperatorConfigs().Update(operatorConfig)
		if err != nil {
 			errs = append(errs, fmt.Errorf("kubecontrollermanageroperatorconfigs/instance: error writing updated observed config: %v", err))
 		} else {
 		    operatorConfig = updatedOperatorConfig
 		}
```

Reply:
interesting... suggestions only works for a single line changes :-)

Reply:
Thanks. In an attempt to explain myself here, I took a closer look at the `Update` api source and realized it wasn't doing what I expected. I've re-worked the logic and added a unit test for the error condition.